src/sagemaker/jumpstart/accessors.py

```diff
@@ -37,6 +38,49 @@ def get_sagemaker_version() -> str:
         return SageMakerSettings._parsed_sagemaker_version
 
 
+class JumpStartS3Accessor(object):
+    """Static class for storing and retrieving auxilliary s3 artifacts."""
+
+    @staticmethod
+    @functools.lru_cache()
+    def _get_default_s3_client(region: str = JUMPSTART_DEFAULT_REGION_NAME) -> boto3.client:
+        """Returns default s3 client associated with the region.
+
+        Result is cached so multiple clients in memory are not created.
+        """
+        return boto3.client("s3", region_name=region)
+
+    @staticmethod
+    @functools.lru_cache()
+    def get_object_cached(
+        bucket: str,
+        key: str,
+        region: str = JUMPSTART_DEFAULT_REGION_NAME,
+        s3_client: Optional[boto3.client] = None,
+    ) -> bytes:
+        """Returns s3 object located at the bucket and key.
+
+        Requests are cached so that the same s3 request is never made more
+        than once, unless a different region or client is used.
+        """
```

Comment:
hum, not sure about this, you are effectively caching objects in memory aren't you?

Could you:
(a) determine the max memory you are willing to use for such cache?
(b) add a head object with a size limits for such objects
(c) derive the max number of items in the `@lru_cache(max_items)` from round((a)/(b)) ?

Reply:
The maximum memory is very system dependent and the payloads will come in all different sizes. How about we expose a function that clears the cache `JumpSartS3Accessor.clear_cache()`?  This can call `JumpSartS3Accessor.get_object_cached.cache_clear()` under the hood. See: https://stackoverflow.com/questions/37653784/how-do-i-use-cache-clear-on-python-functools-lru-cache